### ISSUE TITLE
Improve loading/saving interface

### ISSUE THREAD

**Issue by manics** [manics/jupyter-offlinenotebook#23]
Follow-up from https://github.com/manics/jupyter-offlinenotebook/issues/18

> I think replacing the button to store the notebook with a periodic task (every 30s or so) that saves the current state to IndexedDB would be nice. My hunch would be that people will only remember to press the "save state" button when they try and restore the notebook (aka when it is too late).

This will require a bit of UI work so that we don't overwrite an existing stored copy without checking. A related idea is to proactively offer to load a notebook from storage when a user opens a notebook in a reloaded repository.

**Comment by betatim** [comment]
Good point with the over-writing :-/ I hadn't thought it through that far.

Maybe an alternative way to think about this is that "save/restore from cloud" is an "expert option". In the sense that you have to think about it for a second to understand what is happening and that it requires an explicit action from the user. To help those who only remember to save stuff after it is too late maybe we could display a popup/modal after the kernel connection is gone that says "click here to download the notebook before you close the tab!!".

Needs some pondering about what is easier UI wise maybe.
